int32 ask_cup
---
int32[] cup_result
